core/templates/pages/classroom-admin-page/classroom-admin-page.component.ts

```diff
@@ -144,6 +183,14 @@ export class ClassroomAdminPageComponent implements OnInit {
         this.tempClassroomData = ExistingClassroomData.createClassroomFromDict(
           cloneDeep(response.classroomDict)
         );
+        if (this.tempClassroomData.getThumbnailData().filename) {
+          this.contextService.setCustomEntityContext(
+            AppConstants.ENTITY_TYPE.CLASSROOM,
+            classroomId
+          );
+        } else {
+          this.contextService.removeCustomEntityContext();
+        }
```

Comment:
Could you please explain the need for this logic? Particularly, why would we not require classroom context be set in the classroom admin page.

Reply:
Yes sure. the file path for the saved images looks like this: assetsdevhandler/<entity-type>/<entity-id>/assets/image/<filename>

So in all the other pages like exploration editor, topic editor we can get the entity id from the URL itself but since in the classrooms page the URL does not contain the classroom id that is why we need to set it manually when we click on a classroom tile.

Reply:
Hm, I see but why do we check for the presence of "filename" field in the thumbnail data and then decide to set / unset the custom entity context -- this is the part I am a bit confused about. Please add a comment explaining this. Thanks!

Reply:
The filename check is not needed as curriculum admins should be able to edit banner and thumbnail images independently. Thanks for pointing it out.